### PR TITLE
fix: stop timers when closing TimerEventSources

### LINKED ISSUE #293: Improve unit testing of TimerEventSource
1. reimplement TimerEventSource by uning something else then java.util.Timer, as that one is not good for mocking.
2. implement unit tests without waiting and sleeping to remove flakyness.

### PR DESCRIPTION
**Pull request by lburgazzoli**
Fixes #293

### CODE REVIEW

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/internal/TimerEventSource.java [hunk 1]
```diff
@@ -3,40 +3,53 @@
...
     }
     EventProducerTimeTask task = new EventProducerTimeTask(resourceUid);
     timerTasks.put(resourceUid, task);
     timer.schedule(task, delay, period);
+
+    return true;
   }
 
-  public void scheduleOnce(CustomResource customResource, long delay) {
+  public boolean scheduleOnce(CustomResource customResource, long delay) {
```
Comment (csviri) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/internal/TimerEventSource.java:39]: Is this really a case where the user should check the boolen value? Normally it should not happen that the controlerr is executed before start. So would rather throw a runtime exception in this case.

Comment (lburgazzoli) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/internal/TimerEventSource.java:39]: I did it to make the signature consistent with `schedule(CustomResource,long,long period)` but I can do the same for both the methods

Comment (lburgazzoli) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/internal/TimerEventSource.java]: @csviri migrated to exception

**Comment by lburgazzoli** [comment]
@csviri @adam-sandor given that this PR was ment to fix some issues on the TimerEventSource and that the tests I added are not executed because the related tests were disabled,  I've tried to re-enable tests for the timer event source by replacing the thread sleep with Awaitility

**Review by metacosm — APPROVED** [review]
LGTM

**Comment by lburgazzoli** [comment]
@metacosm anything missing here to get it merged ?

**Comment by metacosm** [comment]
Was just letting more time for other folks to review if they wanted to. 😄

**Comment by lburgazzoli** [comment]
> Was just letting more time for other folks to review if they wanted to. smile

No worries, this is part of my weekly routine to check what I did and what is still ongoing :(

**Pull request merged**